src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxResolvingClassLoader.java

```diff
@@ -26,7 +27,7 @@
     
     private static final Logger LOGGER = Logger.getLogger(SandboxResolvingClassLoader.class.getName());
 
-    private static final LoadingCache<ClassLoader, Cache<String, Optional<Class<?>>>> parentClassCache = makeParentCache();
+    private static final LoadingCache<ClassLoader, Cache<String, Optional<WeakReference<Class<?>>>>> parentClassCache = makeParentCache();
```

Comment:
Why not just use `weakValues` on the inner cache? Is it so we can tell the difference between a value that was not in the cache at all and a value that has been GC'd while in the cache so that we can fall back to the non-cached load? Wouldn't `weakValues` give us that behavior for free since the value would be loaded and put in the cache in both cases?

Reply:
Yes `weakValues` might simplify the patch, I will check it.